File: serial.py
import RPi.GPIO as GPIO
import time

pin = 7
pin1 = 37

def setup():
  GPIO.setmode(GPIO.BOARD)       
  
  GPIO.setup(pin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
  GPIO.setup(pin1, GPIO.OUT)
  
  
  GPIO.output(pin1, GPIO.LOW)
  

def loop():
  while True:
    print(pin, GPIO.input(pin))
    print(pin1, GPIO.input(pin1))
    print()
    time.sleep(0.2)

def destroy():
  GPIO.cleanup()

if __name__ == '__main__':
  setup()
  try:
    loop()
  except KeyboardInterrupt: 
    destroy()

File: serial.js
import { Gpio } from "pigpio";

const HIGH = 1;
const LOW = 0;


function setup() {
    const pin = new Gpio(7, { mode: Gpio.INPUT, pullUpDown: Gpio.PUD_UP });
    const pin1 = new Gpio(37, { mode: Gpio.OUTPUT })

    pin1.digitalWrite(LOW)

    async function loop() {
        console.log("Listening...")
    
        while (true) {
            console.log("7:" + pin.digitalRead())
            console.log("37:" + pin1.digitalRead())
            await new Promise(r => setTimeout(r, 2000));
    
        }
    }
    loop();
}
setup();


